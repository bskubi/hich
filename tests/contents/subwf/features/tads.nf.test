nextflow_process {

    name "Test Process HiCExplorerFindTADs"
    script "contents/subwf/features/tads.nf"
    process "HiCExplorerFindTADs"

    test("findTADs") {

        when {
            process {
                """
                input[0] = tuple(
                    "test", 
                    "$projectDir/tests/assets/Akgol_2021_4DNFI8ZYY7VT_HFF_hic3.0_chr19_100k.mcool", 
                    "100000", 
                    "--minDepth 400000 --maxDepth 1000000 --step 100000 --chromosomes chr19 --correctForMultipleTesting bonferroni"
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
