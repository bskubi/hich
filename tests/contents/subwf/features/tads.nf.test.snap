{
    "findTADs": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "test_boundaries.bed:md5,547c31d808bade66992942bdaf79777f",
                        "test_boundaries.gff:md5,373c2995d81399bba5e9c04bcbd3efbb",
                        "test_domains.bed:md5,7a931ff299d8d779afc2ad54a98a4b4c",
                        "test_score.bedgraph:md5,e4179342b23fe8ce1fadcf871caac433",
                        "test_tad_score.bm:md5,69766b3bf41ebd64f4bf45714707dff2"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-11T19:13:09.902896692"
    }
}